crates/autopilot/src/run.rs

```diff
@@ -373,7 +373,11 @@ pub async fn run(args: Arguments) {
         skip_event_sync_start,
     );
 
-    let mut cow_amm_registry = cow_amm::Registry::new(web3.clone());
+    let cow_amm_web3 = args.archive_node_url.as_ref().map_or(web3.clone(), |url| {
+        boundary::web3_client(url, &args.shared.ethrpc)
+    });
```

Comment:
General question: will cow amm support work properly without an archive node being provided?

Reply:
Yeah, I've worked on that for a while but it turned out a bit ugly so I didn't open a PR for that yet. Should resurrect the work soon. 🤞